yamcs-core/src/main/java/org/yamcs/tctm/cfs/CfsEventDecoder.java

```diff
@@ -131,20 +135,39 @@ public void onTuple(Stream stream, Tuple t) {
         int msgId = ByteArrayUtils.decodeUnsignedShort(packet, 0);
 
         if (msgIds.contains(msgId)) {
+            if (packet.length < MINIMUM_LENGTH) {
+                log.error("Packet must be at least " + MINIMUM_LENGTH + " bytes in size.\n"
+                        + "Packet size:" + packet.length);
+                return;
+            }
             long rectime = (Long) t.getColumn(TM_RECTIME_COLUMN);
             long gentime = (Long) t.getColumn(GENTIME_COLUMN);
 
             try {
                 processPacket(rectime, gentime, packet);
+            } catch (BufferUnderflowException e) {
+                eventProducer.sendWarning("SHORT_PACKET",
+                        "Short packet received, length: " + packet.length);
             } catch (Exception e) {
-                log.warn("Failed to process event packet", e);
+                log.error("Failed to process event packet", e);
             }
         }
     }
 
-    private void processPacket(long rectime, long gentime, byte[] packet) {
+    private void processPacket(long rectime, long gentime, byte[] packet) throws Exception {
         ByteBuffer buf = ByteBuffer.wrap(packet);
+
+        // As per CCSDS protocol, primary header is big endian
+        int expectedLength = ByteArrayUtils.decodeUnsignedShort(packet, 4) + 7;
+
+        if (packet.length < expectedLength) {
+            log.warn("Packet is smaller than the expected length."
```

Comment:
I think you can return here.
And what about generating also an event and mentioning explicitly that the expected length is the length as per the CCSDS header